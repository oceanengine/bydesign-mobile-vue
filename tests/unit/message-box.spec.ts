import Vue from 'vue';
import { mount, later } from '../index';
import { MessageBox as MessageBoxNative, Dialog as DialogNative } from '@src';
const MessageBox = MessageBoxNative as any;
const Dialog = DialogNative as any;

describe('MessageBox', () => {
  test('MessageBox function call', async () => {
    MessageBox.close();
    MessageBox.alert({
      title: 'title',
      message: 'message',
      showCancelButton: true,
    });

    await later();
    const callback = jest.fn();
    const dialog = document.querySelector('.byted-dialog') as HTMLElement;

    expect(dialog.style.display).toEqual('');
    MessageBox.close();
    await later(1000);

    expect(dialog.style.display).toEqual('none');
    MessageBox.confirm().catch(callback);
    (document.querySelector('.byted-dialog__cancel') as HTMLElement).click();

    await later();
    expect(callback).toHaveBeenCalledWith('cancel');
    MessageBox.confirm().then(callback);
    (document.querySelector('.byted-dialog__confirm') as HTMLElement).click();

    await later();
    expect(callback).toHaveBeenNthCalledWith(2, 'confirm');
  });

  test('Dialog custom', async () => {
    const wrapper = mount(Dialog, {
      propsData: {
        value: true,
        title: 'title',
        showCancelButton: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Dialog before close', async () => {
    const wrapper = mount(Dialog, {
      propsData: {
        value: true,
        title: 'title',
        showCancelButton: true,
        closeOnClickOverlay: true,
        beforeClose: (action, done) => {
          done(false);
          action;
        },
      },
    });
    const cancel = wrapper.find('.byted-dialog__cancel');

    cancel.trigger('click');
    expect(wrapper.emitted('input')).toBeFalsy();
    wrapper.setProps({
      beforeClose: (action, done) => {
        if (action === 'cancel') {
          done();
          action;
        }
      },
    });

    const overlay = document.querySelector('.byted-overlay') as HTMLElement;
    overlay.click();
    expect(wrapper.emitted('input')).toBeFalsy();

    cancel.trigger('click');
    expect(wrapper.emitted('input')[0]).toBeTruthy();
  });
  test('register Dialog component', () => {
    Vue.use(Dialog);
    expect(Vue.component(Dialog.name)).toBeTruthy();
  });
  test('default slot', () => {
    const wrapper = mount(Dialog, {
      propsData: {
        value: true,
      },
      scopedSlots: {
        default: () => 'Custom Message',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('title slot', () => {
    const wrapper = mount(Dialog, {
      propsData: {
        value: true,
      },
      scopedSlots: {
        title: () => 'Custom title',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
