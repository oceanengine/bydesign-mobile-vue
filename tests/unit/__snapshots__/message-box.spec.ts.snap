// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox Dialog custom 1`] = `
<transition-stub name="byted-dialog-bounce">
  <div role="dialog" aria-labelledby="title" class="byted-dialog byted-dialog--normal">
    <div class="byted-dialog__header">title</div>
    <div class="byted-dialog__footer byted-hairline--top">
      <div class="byted-button byted-button--default byted-button--normal byted-dialog__cancel">取消</div>
      <div class="byted-button byted-button--default byted-button--normal byted-dialog__confirm byted-hairline--left">确定</div>
    </div>
  </div>
</transition-stub>
`;

exports[`MessageBox default slot 1`] = `
<transition-stub name="byted-dialog-bounce">
  <div role="dialog" class="byted-dialog byted-dialog--normal">
    <div class="byted-dialog__content">Custom Message</div>
    <div class="byted-dialog__footer byted-hairline--top">
      <div class="byted-button byted-button--default byted-button--normal byted-dialog__confirm">确定</div>
    </div>
  </div>
</transition-stub>
`;

exports[`MessageBox title slot 1`] = `
<transition-stub name="byted-dialog-bounce">
  <div role="dialog" class="byted-dialog byted-dialog--normal">
    <div class="byted-dialog__header">Custom title</div>
    <div class="byted-dialog__footer byted-hairline--top">
      <div class="byted-button byted-button--default byted-button--normal byted-dialog__confirm">确定</div>
    </div>
  </div>
</transition-stub>
`;
